pkg/cmd/roachtest/roachtestutil/mixedversion/planner.go

```diff
@@ -75,6 +75,49 @@ type (
 		to             *clusterupgrade.Version
 		sequentialStep sequentialRunStep
 	}
+
+	// mutator describes the interface to be implemented by different
+	// `mutators`.
+	//
+	// Mutators exist as a way to introduce optional behaviour to
+	// upgrade tests in order to increase coverage of certain areas of
+	// the database and of the upgrade machinery itself.
+	mutator interface {
+		// Name returns the unique name of this mutator. This allows the
+		// test runner to log which mutators are active and test authors
+		// to opt-out of mutators that are incompatible with a test, if
+		// necessary.
+		Name() string
+		// Probability returns the probability that this mutator will run
+		// in any given test run. Every mutator should run under some
+		// probability. Making this an explicit part of the interface
+		// makes that more prominent.
+		Probability() float64
+		// Generate takes a test plan and a RNG and returns the list of
+		// mutations that should be applied to the plan.
+		Generate(*rand.Rand, *TestPlan) []mutation
+	}
+
+	// mutationOp encodes the operation to be performed by a mutation.
```

Comment:
Nit: This comment misled me a bit to believe `mutationOp` is the operation itself rather that how it places the mutation operation in relation to a step.